app/rewards/api.py

```diff
@@ -0,0 +1,85 @@
+from datetime import datetime
+from typing import Optional
+import re
+
+from app.common import SimpleMap, BadDataError
+from app.events.service import find_all_reward_events_for_user, generate_referral_event
+from app.auth import with_auth_user
+from app.events.event import Event
+from app.user_api.user_service import find_user_by_referral_code, find_user, update_user
+from app.user_api.errors import UserNotFoundError
+from app.user_api.User import User
+from .errors import ReferralCodeNotFound, InvalidReferralOperation
+
+POINTS_FOR_REFERRAL = 200
+
+
+class RewardEventApiResponse(SimpleMap):
+    def __init__(self, source: str, points: int, timestamp: datetime):
+        self.source = source
+        self.points = points
+        self.timestamp = timestamp
+
+    @staticmethod
+    def from_event(event: Event) -> "RewardEventApiResponse":
+        return RewardEventApiResponse(
+            event.payload.source.value, event.payload.points_change, event.created
+        )
+
+
+@with_auth_user
+def submit_referral_code(user: User, code: str, **kwargs):
```

Comment:
Add a flag to indicate that a user has already used a referral code to prevent abuse

Reply:
Added a new flag on the user `hasUsedReferralCode` that will be set to true once they use their code.